tests/phpunit/tests/post/query.php

```diff
@@ -777,4 +777,525 @@ public function test_found_posts_should_be_integer_even_if_found_posts_filter_re
 
 		$this->assertIsInt( $q->found_posts );
 	}
+
+	/**
+	 * @ticket 47280
+	 */
+	public function test_found_posts_are_correct_for_empty_query() {
+		self::factory()->post->create_many( 12 );
+
+		$q = new WP_Query();
+
+		$this->assertSame( 0, $q->post_count );
+		$this->assertSame( 0, $q->found_posts );
+		$this->assertSame( 0, $q->max_num_pages );
```

Comment:
Tests with multiple assertions ought to have messages to make it easier to determine which assertion causes a test to fail.

```suggestion
		$this->assertSame( 0, $q->post_count, 'Post count is expected to be zero' );
		$this->assertSame( 0, $q->found_posts, 'Total found posts is expected to be zero' );
		$this->assertSame( 0, $q->max_num_pages, 'Number of pages is expected to be zero' );
```

Or similar messaging

🔢 Note: This comment applies in multiple locations but I've only noted it once.